types/dns_zone.pp

```diff
@@ -0,0 +1,2 @@
+# @summary Validate a DNS zone name
+type Stdlib::Dns_zone = Pattern[/\A((([[:alnum:]]|[[:alnum:]][[:alnum:]-]*[[:alnum:]])\.)*[[:alpha:]]+\.|\.)\z/]
```

Comment:
for dns zones i think its even less likley that consumers will want unicode as you are most likley configuering dns software which preferes punycode

> Knot Resolver does not provide any convenience support for IDN. Therefore everywhere (all configuration, logs, RPZ files) you need to deal with the [xn-- forms](https://en.wikipedia.org/wiki/Internationalized_domain_name#Example_of_IDNA_encoding) of domain name labels, instead of directly using unicode characters.
ref: https://knot-resolver.readthedocs.io/en/stable/modules-policy.html#filters

Reply:
Fixed. I also included some punycode in the test suite.